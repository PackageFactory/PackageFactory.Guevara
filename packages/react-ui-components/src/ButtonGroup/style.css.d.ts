export interface IStyleCss {
  'buttonGroup': string;
}

export const locals: IStyleCss;
