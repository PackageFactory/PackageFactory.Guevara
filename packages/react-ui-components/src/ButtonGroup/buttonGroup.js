import React from 'react';
import PropTypes from 'prop-types';
import mergeClassNames from 'classnames';
import ButtonGroupItem from './buttonGroupItem.js';

const ButtonGroup = props => {
    const {
        children,
        className,
        theme,
        value,
        ...rest
    } = props;
    const finalClassName = mergeClassNames({
        [theme.buttonGroup]: true,
        [className]: className && className.length
    });

    return (
        <div {...rest} className={finalClassName}>
            {React.Children.map(children, (child, index) => {
                return (
                    <ButtonGroupItem
                        {...child.props}
                        key={index}
                        onClick={this.props.onSelect}
                        isPressed={value === child.props.id}
                        element={child}
                        />
                );
            })}
        </div>
    );
};
ButtonGroup.propTypes = {
    /**
     * An optional `className` to attach to the wrapper.
     */
    className: PropTypes.string,

    /**
     * Current value
     */
    value: PropTypes.any,

    /**
     * Called when new value is selected. Returns an id of the activated button
     */
    onSelect: PropTypes.func.isRequired,

    /**
     * The contents to be rendered within the `Bar`.
     */
    children: PropTypes.any.isRequired,

    /**
     * An optional css theme to be injected.
     */
    theme: PropTypes.shape({
        buttonGroup: PropTypes.string
    }).isRequired
};

export default ButtonGroup;
