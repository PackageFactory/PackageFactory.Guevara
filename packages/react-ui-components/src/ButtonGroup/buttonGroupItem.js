import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import omit from 'lodash.omit';

class ButtonGroupItem extends PureComponent {
    static propTypes = {
        id: PropTypes.string.isRequired,
        onClick: PropTypes.func.isRequired,
        element: PropTypes.any.isRequired
    };

    handleButtonClick = () => {
        const {onClick, id} = this.props;
        onClick(id);
    }

    render() {
        const {element, ...restProps} = this.props;
        const rest = omit(restProps, ['onClick']);
        return (
            React.cloneElement(element, {
                ...rest,
                onClick: this.handleButtonClick
            })
        );
    }
}

export default ButtonGroupItem;
