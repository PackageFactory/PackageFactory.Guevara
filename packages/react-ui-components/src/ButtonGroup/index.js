import {themr} from '@friendsofreactjs/react-css-themr';
import identifiers from './../identifiers';
import style from './style.css';
import ButtonGroup from './buttonGroup.js';

export default themr(identifiers.buttonGroup, style)(ButtonGroup);
