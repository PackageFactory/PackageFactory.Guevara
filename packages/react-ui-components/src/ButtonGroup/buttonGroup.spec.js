import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import ButtonGroup from './buttonGroup.js';

describe('<ButtonGroup/>', () => {
    let props;

    beforeEach(() => {
        props = {
            theme: {
                btnGroup: 'cleanClassName'
            },
            value: 'foo',
            onSelect: jest.fn(),
            children: [
                <div key="foo" id="foo">Foo button</div>,
                <div key="bar" id="bar">Bar button</div>
            ]
        };
    });

    it('should render correctly.', () => {
        const wrapper = shallow(<ButtonGroup {...props}/>);

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should allow the propagation of "className" with the "className" prop.', () => {
        const wrapper = shallow(<ButtonGroup {...props} className="fooClassName"/>);

        expect(wrapper.prop('className')).toContain('fooClassName');
    });

    it('should allow the propagation of additional props to the wrapper.', () => {
        const wrapper = shallow(<ButtonGroup {...props} foo="bar"/>);

        expect(wrapper.prop('foo')).toBe('bar');
    });
});
