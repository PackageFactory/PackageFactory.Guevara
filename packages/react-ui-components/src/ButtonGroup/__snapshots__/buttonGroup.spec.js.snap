// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonGroup/> should render correctly. 1`] = `
<div
  className=""
>
  <ButtonGroupItem
    element={
      <div
        id="foo"
      >
        Foo button
      </div>
    }
    id="foo"
    isPressed={true}
    key="0/.$foo"
    onClick={[MockFunction]}
  >
    Foo button
  </ButtonGroupItem>
  <ButtonGroupItem
    element={
      <div
        id="bar"
      >
        Bar button
      </div>
    }
    id="bar"
    isPressed={false}
    key="1/.$bar"
    onClick={[MockFunction]}
  >
    Bar button
  </ButtonGroupItem>
</div>
`;
